app/mm-redux/reducers/entities/channels.ts

```diff
@@ -11,6 +11,7 @@ import {Team} from '@mm-redux/types/teams';
 function removeMemberFromChannels(state: RelationOneToOne<Channel, UserIDMappedObjects<ChannelMembership>>, action: GenericAction) {
     const nextState = {...state};
     Object.keys(state).forEach((channel) => {
+        nextState[channel] = {...nextState[channel]};
```

Comment:
shouldn't this be cloned already on line 12?

Reply:
Not really, in the line 12 you are cloning all the "references" to the objects, but not the objects itself, this way we are cloning the references in first place and then the specific channels.